embedded-graphics/src/image/mod.rs

```diff
@@ -1,33 +1,209 @@
-//! Image drawables.
-//!
-//! Image drawables can be created for raw bitmap data and images in BMP and TGA
-//! format.
+//! Image support for embedded-graphics
 
-#[cfg(feature = "bmp")]
-mod image_bmp;
 mod image_raw;
-#[cfg(feature = "tga")]
-mod image_tga;
 
-pub use self::image_raw::{Image, ImageBE, ImageLE};
+pub use self::image_raw::{ImageBE, ImageLE, ImageRaw};
 
-#[cfg(feature = "bmp")]
-pub use self::image_bmp::ImageBmp;
-#[cfg(feature = "tga")]
-pub use self::image_tga::ImageTga;
+use crate::{
+    draw_target::DrawTarget,
+    drawable::{Drawable, Pixel},
+    geometry::{Dimensions, Point, Size},
+    pixelcolor::PixelColor,
+    transform::Transform,
+};
+use core::marker::PhantomData;
 
-/// Image file trait.
-pub trait ImageFile<'a>: crate::geometry::Dimensions + Sized {
-    /// Create a new image with given input file
-    ///
-    /// The input file is expected to be of a particular format (BMP, TGA, etc) and contain file
-    /// metadata like width/height and pixel data. Because parsing may fail, this returns a
-    /// `Result<Self, ()>`.
-    fn new(filedata: &'a [u8]) -> Result<Self, ()>;
+/// Image data trait
+///
+/// This trait is the main integration point for image loading crates. Images are made drawable
+/// through use of the [`Image`] struct which accepts an `ImageData`. `Image` implements
+/// [`Drawable`], allowing it to be drawn on any display that supports embedded-graphcis via the
+/// [`DrawTarget`] trait.
+///
+/// [`DrawTarget`]: ../trait.DrawTarget.html
+/// [`Drawable`]: ../drawable/trait.Drawable.html
+/// [`Image`]: ./struct.Image.html
+pub trait ImageData<C>
+where
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    /// Iterator over pixels in the image
+    type PixelIterator: Iterator<Item = Pixel<C>>;
 
     /// Get the width in pixels of an image
     fn width(&self) -> u32;
 
     /// Get the height in pixels of an image
     fn height(&self) -> u32;
+
+    /// Get an iterator over the pixels of the image
+    fn pixel_iter(&self) -> Self::PixelIterator;
+}
+
+/// A wrapper for any image type
+///
+/// This is a wrapper around [`ImageData`] implementations to better integrate into
+/// embedded-graphics.
+///
+/// [`ImageData`]: ./trait.ImageData.html
+#[derive(Debug, Clone, Copy)]
+pub struct Image<I: Clone, C> {
+    image: I,
+    offset: Point,
+    c: PhantomData<C>,
+}
+
+impl<'a, I, C> Image<I, C>
+where
+    I: ImageData<C> + Clone,
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    /// Create a new `Image` with a given [`ImageData`]
+    ///
+    /// The passed [`ImageData`] provides a source of pixel data from the original image.
+    ///
+    /// [`ImageData`]: ./trait.ImageData.html
+    pub fn new(image: I) -> Self {
```

Comment:
The `Text::new` function has an additional `position` parameter. IMO it would be a good idea to also add this parameter here.